### ISSUE TITLE
[Feature Request] Highlight the *Git* directory root differently

### ISSUE THREAD

**Issue by stevenxxiu** [romkatv/powerlevel10k#1578]
I've always wanted to know which directory is the root of my current *Git* directory.

It seems to me, that *Git* directories are highlighted in bold:

![Screenshot_2021-09-19_00-53-53](https://user-images.githubusercontent.com/2058940/133893012-b9225429-f236-48f3-8dfe-cc7935b68cf7.png)

However the directory I'm current in is also highlighted in bold:

![Screenshot_2021-09-19_00-55-46](https://user-images.githubusercontent.com/2058940/133893035-47e45282-e770-4cfe-ada8-40aead1f2319.png)

I have my home directory under version control as well. As such, it's difficult to tell which directory is the *Git* directory root.

Is there a way to highlight the *Git* directory root differently?

**Comment by romkatv** [comment]
> Is there a way to highlight the Git directory root differently?

There isn’t .

**Comment by stevenxxiu** [comment]
Thanks for the info. I'll change this to a feature request then.

**Comment by stevenxxiu** [comment]
> For the sake of aligning expectations: I don't intend to work on this feature request.

That's ok. Could you perhaps tell me where in the code the directory parts are highlighted in bold? I'll see if I can get it working and send in a PR.

**Comment by romkatv** [comment]
Search for "ANCHOR_BOLD" in https://github.com/romkatv/powerlevel10k/blob/master/internal/p10k.zsh. This code is not for the fainthearted.

**Comment by romkatv** [comment]
I assume you've already looked at https://github.com/romkatv/powerlevel10k/blob/4f3d2ffe7251a17aa10e00ed30d55f6fd9d65002/config/p10k-rainbow.zsh#L218-L261 and found out that you cannot achieve what you want by tweaking these parameters. If not, do try that. You might also want to try `POWERLEVEL9K_DIR_PATH_HIGHLIGHT_FOREGROUND` and `POWERLEVEL9K_DIR_PATH_HIGHLIGHT_BOLD`. These allow you to style the last segment independently from anchors.

**Comment by stevenxxiu** [comment]
Ah thanks. I wasn't aware of those settings. That actually largely does what I want, as my anchor directories are probably ones under version control anyway.

The only issue is that I can't really tell if my current directory is under version control or not. I would like to use the anchor color if the current directory is an anchor folder.

**Comment by romkatv** [comment]
> The only issue is that I can't really tell if my current directory is under version control or not. I would like to use the anchor color if the current directory is an anchor folder.

Sounds like `POWERLEVEL9K_SHORTEN_DIR_LENGTH=0`. The rest of the parameters can stay at their default values as I linked above.

**Comment by stevenxxiu** [comment]
> Sounds like `POWERLEVEL9K_SHORTEN_DIR_LENGTH=0`. The rest of the parameters can stay at their default values as I linked above.

Thanks so much! That actually solves my issues. The only small annoyance ig is the current directory isn't highlhighted in bold anymore, if we're not in an anchor directory. But guess that's not too bad.

My settings are:

    typeset -g POWERLEVEL9K_DIR_ANCHOR_FOREGROUND=220
    ...
    typeset -g POWERLEVEL9K_SHORTEN_DIR_LENGTH=0

**Comment by romkatv** [comment]
> The only small annoyance ig is the current directory isn't highlighted in bold anymore, if we're not in an anchor directory.

Try `POWERLEVEL9K_DIR_PATH_HIGHLIGHT_BOLD=true`.

**Comment by romkatv** [comment]
By the way, the awfulness of these parameters' names is only partially my fault. Backward compatibility with Powerlevel9k has its downsides.

**Comment by stevenxxiu** [comment]
> Try `POWERLEVEL9K_DIR_PATH_HIGHLIGHT_BOLD=true`.

I did try that with my settings, but this overrides the anchor's color. So then I can't tell if the directory is an anchor or not.

**Comment by romkatv** [comment]
I see. The current code explicitly favors the styling of the last segment if it's specified and ignores the styling of the anchor. I don't remember why I implemented it this way but I this this can be changed so that the two styles combine. It's this part: https://github.com/romkatv/powerlevel10k/blob/4f3d2ffe7251a17aa10e00ed30d55f6fd9d65002/internal/p10k.zsh#L2057-L2062

That code in plain English:

```text
if (no special styling for the last segment is required) {
  apply anchor styling to all anchors
} else {
  apply anchor styling to all anchors except the last segment
}
```

**Comment by stevenxxiu** [comment]
Ah do you think I could do a PR for that?

**Comment by romkatv** [comment]
If you send a clean PR, I'll merge it. It may not be easy though. If you believe you have strong skills w.r.t. zsh scripting (e.g., you would be the best among 10 random zsh users), it should be doable.

(My personal advice is to not do it. The benefit is small even for you, and it'll likely be zero for everyone else. No one uses `HIGHLIGHT` parameters.)
